wn/lmf.py

```diff
@@ -90,12 +99,31 @@ def __iter__(self):
     def __next__(self):
         _next = self._next
         if _next == (None, None):
+            if self._progress_info:
+                self._progress_info.progress_handler.set(status="Completed")
             raise StopIteration
         self._next = next(self.iterator, (None, None))
         return _next
 
+    def _update_progress(self):
+        if self._progress_info:
+            self._progress += 1
+            # We don't want to refresh each cycle
+            refresh_cycle = 20
+            if self._progress % refresh_cycle == 0:
+                if self._progress < self._progress_info.total_items:
+                    self._progress_info.progress_handler.update(refresh_cycle)
+                    self._refresh_count += refresh_cycle
+                else:
+                    last_refresh = self._refresh_count
+                    last_refresh = self._progress_info.total_items - last_refresh
+                    self._progress_info.progress_handler.update(last_refresh)
+
     def starts(self, *tags: str) -> bool:
         event, elem = self._next
+        if "LexicalEntry" in tags:
+            self._update_progress()
```

Comment:
Again, I think this should be updated for more than just lexical entries.

Reply:
addressed